rest/route/annotations.go

```diff
@@ -264,8 +265,33 @@ func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request)
 		}
 	}
 
+	if taskExecutionsAsString != "" {
+		taskExecution, err := strconv.Atoi(taskExecutionsAsString)
+		if err != nil {
+			return gimlet.ErrorResponse{
+				Message:    "cannot convert execution to integer value",
+				StatusCode: http.StatusBadRequest,
+			}
+		}
+
+		if h.annotation.TaskExecution == nil {
+			h.annotation.TaskExecution = &taskExecution
+		} else if *h.annotation.TaskExecution != taskExecution {
+			return gimlet.ErrorResponse{
+				Message:    "Task execution must equal the task execution specified in the annotation",
+				StatusCode: http.StatusBadRequest,
+			}
+		}
+	}
+
+	if h.annotation.TaskExecution == nil && taskExecutionsAsString == "" {
```

Comment:
At this point, h.annotation.TaskExecution because of the assignment on `278` will already be equal to either what was in the body or the url. So you should be able to just check if ` h.annotation.TaskExecution == nil`.

Reply:
h.annotation.TaskExecution will only be set if the execution string from the url is not empty so line 278 would not be hit if the string is empty.  This line serves to throw an error only if both fields are never set so I think this boolean statement should remain.

Reply:
We do still need the boolean statement, but I'm wondering if we need both sides of the && in order to check "has the user not specified it anywhere" considering the check on 277.

Reply:
> We do still need the boolean statement, but I'm wondering if we need both sides of the && in order to check "has the user not specified it anywhere" considering the check on 277.

Simplified the statement